pkg/registry/reference.go

```diff
@@ -59,6 +61,22 @@ func ParseReference(s string) (*Reference, error) {
 	return &ref, nil
 }
 
+// ParseReferenceWithChartDefaults converts a string to a Reference,
+// using values from a given chart as defaults
+func ParseReferenceWithChartDefaults(s string, ch *chart.Chart) (*Reference, error) {
+	ref, err := ParseReference(s)
+	if err != nil {
+		return nil, err
+	}
+
+	// If no tag is present, use the chart version
```

Comment:
instead of introducing a new function, can we just add this conditional logic to pkg/action/chart_save.go? e.g.

```
r, err := registry.ParseReference(ref)
	r, err := registry.ParseReferenceWithChartDefaults(ref, ch)
	if err != nil {
		return err
	}

	if r.Tag == "" {
		r.Tag = ch.Metadata.Version
	}

	...
```

Reply:
I originally had done this, but decided against it for a few reasons:

a.) it wasn't a change that was easily tested via action package as the method had no returns besides "err" - unless we run a test against the stdout, which seems a little bleh. This new method is easy to verify via tests

b.) it occurred to me that we may want to add more transform features based on the chart, for example `myreg.io/mylib/@name` might replace "@name" with chart name? Something along these lines

c.) we may want to get this functionality somewhere outside of the save method

@bacongobbler what do you think? I'm happy to revert - but need some ideas on how to correctly unit test this if so

Reply:
taking a closer deep dive into the code now :eyes: